src/api/form/PDFForm.ts

```diff
@@ -499,6 +508,68 @@ export default class PDFForm {
     return PDFTextField.of(text, text.ref, this.doc);
   }
 
+  /**
+   * Flatten all form fields.
+   * 
+   * Flattening a form field will take the current appearance and make that part
+   * of the pages content stream. All form fields and annotations associated are removed.
+   * 
+   * For example:
+   * ```js
+   * const form = pdfDoc.getForm();
+   * form.flatten();
+   * ```
+   */
+  flatten() {
+    this.updateFieldAppearances();
+
+    const fields = this.getFields();
+    const pages = this.doc.getPages();
+
+    for (const field of fields) {
+      const widgets = field.acroField.getWidgets();
+
+      for (const widget of widgets) {
+        const pageRef = widget.P();
+        const page = pages.find(x => x.ref === pageRef);
```

Comment:
We should assert that we did, in fact, find a match. This will also remove the need for non-null assertions later in the method.
```suggestion
        const page = pages.find(x => x.ref === pageRef);
        if (!page) throw new Error(`Failed to find page: ${pageRef}`);
```